test/browser/spec/ol/style/icon.test.js

```diff
@@ -419,14 +441,27 @@ describe('ol.style.Icon', function () {
       });
       expect(iconStyle.getWidth()).to.eql(10);
     });
-    it('setWidth updates the width', function () {
+    it('setWidth updates the width, and scale when image is loaded', function (done) {
       const iconStyle = new Icon({
-        src,
-        width: 10,
+        src: img,
+        width: 9,
```

Comment:
What if you leave src and width unchanged here?

Reply:
The src has a known width and 9 avoids a fractional scale.  Not important when testing before loading but needed to get results after loading.